cmds/ntpd/ntpd.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// simple ntpd daemon in Go.
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/beevik/ntp"
+)
+
+var (
+	config  = flag.String("config", "/etc/ntp.conf", "NTP config file.")
+	verbose = flag.Bool("verbose", false, "Verbose output")
+	debug   = func(string, ...interface{}) {}
+)
+
+const (
+	fallback = "time.google.com"
+)
+
+func parseServers(r *bufio.Reader) []string {
+	var uri []string
+	var l string
+	var err error
+
+	debug("Reading config file\n")
+	for err == nil {
+		// This handles the case where the last line doesn't end in \n
+		l, err = r.ReadString('\n')
+		debug("%v", l)
+		if w := strings.Fields(l); len(w) > 1 && w[0] == "server" {
+			// We look only for the server lines, we ignore options like iburst
+			// TODO(ganshun): figure out what options we want to support.
+			uri = append(uri, w[1])
+		}
+	}
+
+	return uri
+}
+
+func getTime(servers []string) (t time.Time, err error) {
+	for _, s := range servers {
+		debug("Getting time from %v\n", s)
+		if t, err = ntp.Time(s); err == nil {
+			// Right now we return on the first valid time.
+			// We can implement better heuristics here.
+			debug("Got time %v\n", t)
+			return
+		}
+		debug("Error getting time: %v\n", err)
+	}
+
+	t = time.Now()
```

Comment:
why bother with this extra work? t will have the zero value. You should never use it in the case of an error.

Reply:
you could also have it return t *time.Time and then return nil, err if it's an error?

Reply:
I'll change it to return nil